### PR TITLE
tdm: removed

### LINKED ISSUE #98936: tdm: The Dark Mod fails to start on NixOS 20.09
**Describe the bug**

**To Reproduce**
1. Install pkgs.tdm
2. Execute tdm_updater
3. Execute tdm

**Expected behavior**
tdm starts.

**Actual behaviour**

tdm starts with a black error and has stderr errors such as:
```
----- R_ReloadARBPrograms -----
WARNING:LoadARBProgram: 'glprogs/test.vfp' not found
WARNING:LoadARBProgram: 'glprogs/test.vfp' not found
WARNING:LoadARBProgram: 'glprogs/interaction.vfp' not found
WARNING:LoadARBProgram: 'glprogs/interaction.vfp' not found
glprogs/bumpyEnvironment.vfp 5
glprogs/bumpyEnvironment.vfp 6
WARNING:LoadARBProgram: 'glprogs/ambientLight.vfp' not found
WARNING:LoadARBProgram: 'glprogs/ambientLight.vfp' not found
WARNING:LoadARBProgram: 'glprogs/shadow.vp' not found
glprogs/environment.vfp 3
glprogs/environment.vfp 4
WARNING:LoadARBProgram: 'glprogs/test_direct.vfp' not found
WARNING:LoadARBProgram: 'glprogs/test_direct.vfp' not found
WARNING:LoadARBProgram: 'glprogs/interaction_direct.vfp' not found
WARNING:LoadARBProgram: 'glprogs/interaction_direct.vfp' not foun
```

**Additional context**

It looks like the problem might be that pkgs.tdm and the assets downloaded via tdm_updater are different versions:

```
$ ls -lah $(which tdm)
lrwxrwxrwx 1 root root 60 Jan  1  1970 /run/current-system/sw/bin/tdm -> /nix/store/izddmzwjwv6aaim6kq65lbyax4qp6qf5-tdm-2.07/bin/tdm
$ tail ~/.local/share/tdm/tdm_version_info.txt 
filesize = 17967512

[Version2.08 File thedarkmod.x86]
crc = 38ef5a25
filesize = 18517512

[Version2.08 File TheDarkModx64.exe]
crc = 160957d2
filesize = 19812352
```

** Workaround

`cd ~/.local/share/tdm && steam-run ./thedarkmod.x64`

(Steam isn't actually used for this. This is just a lazy way to run FHS programs.)

**Notify maintainers**
@cizra 

**Metadata**

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: tdm
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by cizra**
###### Motivation for this change
I can't get sound to work in The Dark Mod - OpenAL is not detecting any hardware.
As a workaround, it's possible to use a precompiled version with steam-run.
Fixes  [#98936](https://github.com/NixOS/nixpkgs/issues/98936)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please remove it from ``pkgs/top-level/all-packages.nix`` and add an alias that throws that it got removed.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -28562,7 +28562,10 @@ in
     gcc-armhf-embedded = pkgsCross.armhf-embedded.buildPackages.gcc;
   };
 
-  tdm = callPackage ../games/tdm { };
+  tdm = throw (
+    "tdm has been removed because nobody can figure out how to fix OpenAL integration." +
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:28566]: This belongs into aliases.nix.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Also please resolve the merge conflict.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (cizra) [reply pkgs/top-level/all-packages.nix]: Fixed.

**Review by SuperSandro2000 — APPROVED** [review]
I think this is fine now.

**Pull request merged**